### PR TITLE
Bump go version to 1.17.8

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
Bumps go version to 1.17.8
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Shoot oidc service extension is now built using `golang:1.17.8`.
```

### CODE REVIEW

File: .test-defs/lifecycle.yaml [hunk 1]
```diff
@@ -18,4 +18,4 @@ spec:
     --shoot-name=$SHOOT_NAME
     --project-namespace=$PROJECT_NAMESPACE
     --kubecfg="$TM_KUBECONFIG_PATH/gardener.config"
-  image: golang:1.17.5
+  image: golang:1.17.7
```
Comment (vpnachev) [inline .test-defs/lifecycle.yaml:21]: Any objections against the latest golang version?
```suggestion
  image: golang:1.17.8
```

Comment (dimityrmirchev) [reply .test-defs/lifecycle.yaml]: Nope

**Review by vpnachev — APPROVED** [review]
/lgtm
/title Bump go version to 1.17.8

**Pull request merged**